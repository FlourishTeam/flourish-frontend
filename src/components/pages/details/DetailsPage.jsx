import React from 'react'

function DetailsPage() {
    return (
        <div>
            Hello I am Details.
        </div>
    )
}

export default DetailsPage
