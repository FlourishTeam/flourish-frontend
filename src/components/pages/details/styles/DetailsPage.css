@import url('https://fonts.googleapis.com/css2?family=Kavivanar&display=swap');
@import url('https://fonts.googleapis.com/css2?family=PT+Sans&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Salsa&display=swap');

@media screen and (min-width: 1001px) {
    .careDetails {
        font-size: 1em;
    }
}

@media screen and (max-width: 1000px) {
    .careDetails {
        font-size: .7em;
    }
    .bottomMidRight {
        font-size: .5em;
        line-height: 120%;
    }
    .bottomMidLeft {
        font-size: .8em;
        line-height: 120%;
    }
}

.Details {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    background-image: linear-gradient(to top, #25282a 60%, #111411 100%);
    color: white;
}

.top {
    display: flex;
    flex-direction: column;
    width: 40%;
    margin-top: 3em;
    text-align: center;
    font-family: 'Salsa', cursive;
    letter-spacing: .2em;
}

.common {
    font-size: 3em;
    color: white;
}

.scientific {
    font-style: italic;
    font-weight: normal;
    color: white;
}

.maintenanceValue {
    float: right;
    position: relative;
}

.mid {
    border: 4px solid black;
    border-radius: 10px;
    margin: 2em;
    height: 35em;
    width: 90%;
    background-color: white;
    color: #022704;
    box-shadow: 0px 2px 10px 2px black;
}

.midLeft {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    float: left;
    width: 35%;
    height: 100%;
    padding: 1em;
    margin-left: 2.5em;
}

.image {
    width: auto;
    height: auto;
}

.addRemoveContainer {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin-top: 1em;
}

.midRight {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    vertical-align: middle;
    height: 100%;
}

.careDetails {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    width: 90%;
    height: 100%;
    border-left: 3px solid black;
    line-height: 135%;
     font-family: 'PT Sans', sans-serif;
}

.key {
    float: left;
    width: 35%;
    text-align: right;
}

.value {
    float: right;
    width: 60%;
}

.bottomMid {
    display: inline-block;
    justify-content: space-evenly;
    align-items: center;
    width: 90%;
    height: 15em;
    border-radius: 10px;
    line-height: 135%;
    padding: 0 2em 0 2em;
    background-color: black;
    color: #EBFEF4;
    border: 1px solid white;
    font-family: 'PT Sans', sans-serif;
    box-shadow: 0px 2px 10px 2px black;
}

.bottomMidTop {
    width: 100%;
    text-align: center;
    font-size: 1.5em;
}

.bottomMidLeft {
    float: left;
    text-align: center;
    width: 30%;
}

ul {
    list-style-type: none;
}

.bottomMidRight {
    float: right;
    width: 60%;
}

.title {
    font-weight: bold;
    font-size: 1.2em;
    padding: 1em;
}

.propagation {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
    width: 90%;
    line-height: 135%;
    margin: 2em 0 2em 0;
    padding: 2em 0 2em 0;
    border: 4px solid black;
    border-radius: 10px;
    background-color: white;
    color: #022704;
    font-family: 'PT Sans', sans-serif;
    box-shadow: 0px 2px 10px 2px black;
}

.propKey {
    font-weight: bold;
    font-size: 1.5em;
}

.loadingContainer {
    margin: 10%;
    display: flex;
    justify-content: center;
    align-items: center;
}