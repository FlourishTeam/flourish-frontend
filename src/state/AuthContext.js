import React, { createContext, useContext, useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { getVerify, postLogin, postSignup } from '../services/auth';

const AuthContext = createContext(null);

export const AuthProvider = ({ children }) => {
  const history = useHistory();
  const [session, setSession] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);
  const isAuthenticated = !!session;

  useEffect(() => {
    getVerify()
      .then(user => setSession(user))
      .catch(() => console.log('user not logged in'))
      .finally(() => setLoading(false));
  }, []);

  const signup = (email, password) => {
    return postSignup(email, password)
      .then(user => setSession(user))
      .then(() => history.push('/'))
      .catch(err => setError(err));
  };

  const login = (email, password) => {
    return postLogin(email, password)
      .then(user => setSession(user))
      .then(() => history.push('/'))
      .catch(err => setError(err));
  };

  return (
    <AuthContext.Provider value={{
      session,
      loading,
      error,
      isAuthenticated,
      signup,
      login
    }}>{children}</AuthContext.Provider>
  );
};

export const useSession = () => {
  const { session } = useContext(AuthContext);
  return session;
};

export const useAuthLoading = () => {
  const { loading } = useContext(AuthContext);
  return loading;
};

export const useAuthError = () => {
  const { error } = useContext(AuthContext);
  return error;
};

export const useIsAuthenticated = () => {
  const { isAuthenticated } = useContext(AuthContext);
  return isAuthenticated;
};

export const useSignup = () => {
  const { signup } = useContext(AuthContext);
  return signup;
};

export const useLogin = () => {
  const { login } = useContext(AuthContext);
  return login;
};


