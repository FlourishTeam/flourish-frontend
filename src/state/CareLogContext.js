import React, { createContext, useContext, useState } from 'react';
import client from './GraphQLContext';
import { gql } from 'apollo-boost';

export const CareLogContext = createContext(null);

export const CareLogProvider = ({ children }) => {
  const [loading, setLoading] = useState(true);
  const [plantDetails, setPlantDetails] = useState({});
  const [careLogItems, setCareLog] = useState(null);
  const [error, setError] = useState(null);

  const renderMyCareHistory = (userId, plantId) => {
    setError(null);
    setLoading(true);

    client
      .query({
        query: gql`
        query {
          getMyCareHistoryById(plantId: ${plantId}, userId: ${userId}) {
            careLogs {
              userPlantLogId
              userId
              plantId
              userPlantId
              careDate
              careType
              careNote
            }
            plantDetails {
              plantId
              image
              commonName
              synonyms
              scientificName
              lightRange
              hydrationRange
              careDifficulty
              pestsDiseases
              warnings
              height
              spread
              type
              floweringPeriod
              bloomSize
              temperatureRange
              placement
              substrateRecommendation
              pottingNotes
              watering
              propagation
            }
          }
        }
      `, })
      .then(({ data }) => {
        setCareLog(data.getMyCareHistoryById);
        setPlantDetails(data.plantDetails);
        setLoading(false);
      })
      .catch((error) => setError(error.message));
  };


  return (
    <CareLogContext.Provider value={{ 
      loading,
      careLogItems,
      renderMyCareHistory,
      plantDetails,
      error 
    }}>
      {children}
    </CareLogContext.Provider>
  );
};

export const useCareLogLoading = () => {
  const { loading } = useContext(CareLogContext);
  return loading;
};

export const useCareLogItems = () => {
  const { careLogItems } = useContext(CareLogContext);
  return careLogItems;
};

export const usePlantDetails = () => {
  const { plantDetails } = useContext(CareLogContext);
  return plantDetails;
};

export const useCareLogError = () => {
  const { error } = useContext(CareLogContext);
  return error;
};

export const useRenderMyCareHistory = () => {
  const { renderMyCareHistory } = useContext(CareLogContext);
  return renderMyCareHistory;
};
