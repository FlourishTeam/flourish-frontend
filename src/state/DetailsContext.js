import React, { createContext, useContext, useState } from 'react';
import client from './GraphQLContext';
import { gql } from 'apollo-boost';

export const DetailsContext = createContext(null);

export const DetailsProvider = ({ children }) => {
  const [loading, setLoading]  = useState(true);
  const [plant, setPlant] = useState({});
  const [error, setError] = useState(null);

  const renderDetails = (id) => {
    setError(null);
    setLoading(true);

    client
      .query({
        query: gql`
        query {
          plantById(id: ${id}) {
          plantId,
          image,
          commonName,
          scientificName,
          lightRange,
          hydrationRange,
          temperatureRange,
          placement,
          substrateRecommendation,
          pottingNotes,
          watering,
          synonyms,
          pestsDiseases,
          warnings,
          height,
          spread,
          type,
          floweringPeriod,
          bloomSize,
          propagation,
          careDifficulty
          }
        }
        `, })
      .then(({ data }) => {
        setPlant(data.plantById);
        setLoading(false);
        console.log('DETAILS CONTEXT PLANT', plant);
      })
      .catch((error) => setError(error.message));
  };
  
  return (
    <DetailsContext.Provider value={{ 
      loading, 
      plant, 
      error,
      renderDetails }}>
      {children}
    </DetailsContext.Provider>
  );
};

export const useDetailsLoading = () => {
  const { loading } = useContext(DetailsContext);
  return loading;
};

export const usePlant = () => {
  const { plant } = useContext(DetailsContext);
  return plant;
};

export const useDetailsError = () => {
  const { error } = useContext(DetailsContext);
  return error;
};

export const useRenderDetails = () => {
  const { renderDetails } = useContext(DetailsContext);
  return renderDetails;
};
