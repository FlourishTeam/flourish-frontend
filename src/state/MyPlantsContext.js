import React, { createContext, useContext, useState } from 'react';
import { data } from '../hooks/fake-data';


export const MyPlantContext = createContext(null);

export const MyPlantProvider = ({ children }) => {
  const [loading, setLoading] = useState(true);
  const [myPlants, setMyPlants] = useState(data);

  return (
    <MyPlantContext.Provider value={{ myPlants, loading }}>
      {children}
    </MyPlantContext.Provider>
  );
};

export const useMyPlants = () => {
  return useContext(MyPlantContext);
};
