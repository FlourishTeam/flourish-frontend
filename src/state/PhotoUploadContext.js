import React, { createContext, useContext, useState } from 'react';
import { useHistory } from 'react-router-dom';
import {
  useHandleSearch,
  useSearchLoading,
  useSetSearchLoading,
} from './SearchContext';

const API_URL = process.env.REACT_APP_API_URL;
const API_KEY = process.env.REACT_APP_API_KEY;

export const PhotoUploadContext = createContext(null);

export const PhotoUploadProvider = ({ children }) => {
  const [photoMode, setPhotoMode] = useState(false);
  const [pictureUpload, setPictureUpload] = useState(
    'images/placeholder-img.png'
  );
  const [empty, setEmpty] = useState(true);
  const handleSearch = useHandleSearch();
  const setLoading = useSetSearchLoading();

  const history = useHistory();

  const handlePreview = (e) => {
    setPictureUpload(URL.createObjectURL(e.target.files[0]));
    setEmpty(false);
  };

  const handleUpload = (e) => {
    e.preventDefault();
    const files = [...e.target.elements.uploader.files];

    if (files.length > 0) {
      setLoading(true);

      const promise = files.map((file) => {
        return new Promise((resolve, reject) => {
          const reader = new FileReader();
          reader.onload = (e) => {
            const res = e.target.result;
            resolve(res);
          };
          reader.readAsDataURL(file);
        });
      });

      Promise.all(promise).then((encodedStr) => {
        const data = {
          api_key: `${API_KEY}`,
          images: encodedStr,
        };

        fetch(`${API_URL}`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data),
        })
          .then((res) => res.json())
          .then((json) => {
            handleSearch(json.suggestions[0].plant_name);
          });
      });

      setPhotoMode(false);
      history.push('/search');
    }
  };

  return (
    <PhotoUploadContext.Provider
      value={{
        photoMode,
        setPhotoMode,
        pictureUpload,
        handleUpload,
        handlePreview,
        empty,
      }}
    >
      {children}
    </PhotoUploadContext.Provider>
  );
};

export const usePhotoUploader = () => {
  return useContext(PhotoUploadContext);
};
