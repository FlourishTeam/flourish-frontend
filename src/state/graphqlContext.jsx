import fetch from 'cross-fetch';
import { ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client';

const httpLink = createHttpLink({
  uri: 'https://localhost:7891/graphql',
  fetch,
});
const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
});
  
export default client;
