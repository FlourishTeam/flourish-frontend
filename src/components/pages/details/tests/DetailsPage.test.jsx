import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router';
import { DetailsProvider } from '../../../../state/DetailsContext';
import DetailsPage from '../DetailsPage';

describe('DetailsPage component', () => {
  afterEach(() => cleanup());
  it('renders DetailsPage', () => {
    const { asFragment } = render(
      <Router>
        <DetailsProvider>
          <DetailsPage />
        </DetailsProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
