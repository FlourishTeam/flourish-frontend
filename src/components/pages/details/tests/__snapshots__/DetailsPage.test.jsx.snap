// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component renders DetailsPage 1`] = `
<DocumentFragment>
  <img
    alt="Loading"
    src="http://www.jainstechnology.com/wp-content/uploads/2018/02/loader.gif"
    styles="[object Object]"
  />
</DocumentFragment>
`;
